pkg/prometheus/promcfg.go

```diff
@@ -2998,12 +2998,10 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		configs := make([][]yaml.MapItem, len(sc.Spec.EC2SDConfigs))
 		for i, config := range sc.Spec.EC2SDConfigs {
 			if config.Region != nil {
-				configs[i] = []yaml.MapItem{
-					{
-						Key:   "region",
-						Value: config.Region,
-					},
-				}
+				configs[i] = append(configs[i], yaml.MapItem{
```

Comment:
hmm nice catch! we had a latent bug here?

Reply:
Yeah, I noticed that the Filters were not getting added, turns out this is the reason.